app/models/genre.rb

```diff
@@ -15,4 +15,12 @@ class Genre < ApplicationRecord
   validates :name, presence: true, uniqueness: true
 
   normalized_col_generator :name
+
+  def merge(other)
+    tracks = self.tracks
+    tracks.find_each do |track|
```

Comment:
Why update via tracks? Can't you just add all tracks of the other genre to this genre? I think/hope that would make for a lot less database calls.

Reply:
Ignore this, it doesn't seem like significantly reducing the number of database calls is possible, and using the `find_each` over tracks will result in less memory usage. I do worry that the action in the controller might take a very long time though, if a genre with a **lot** of tracks is merged.